### PR TITLE
refactor(stark-demo): refactor project folder structure of Showcase and Starter. Adapt documentation.

### PR DESCRIPTION
**Pull request by christophercr**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The application structure of both the Showcase and the Starter was not consistent and sometimes confusing.

## What is the new behavior?
The structure has been refactored and adapted to what we had in Stark 9 and also following the [Angular styleguide](https://angular.io/guide/styleguide#folders-by-feature-structure). This is documented in the `README.md` of the Showcase and the Starter 😉 

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other
Took the change to re-generate all the `package-lock.json` files of all the packages

### CODE REVIEW

**Comment by christophercr** [comment]
@Mallikki @cnomes @SuperITMan can you have a quick look on the `README.md` file of the Showcase and see if it is ok? There is explained everything what this change is all about. The `README.md` in the Starter has the same modifications.

File: showcase/README.md [hunk 1]
```diff
@@ -2,96 +2,149 @@
...
+|   |   |   |   \---...
+|   |   |   |
+|   |   |   +---actions                              # NGRX actions utility classes for this module
+|   |   |   |       index.ts                         # barrel: re-export all actions
+|   |   |   |       foo.actions.ts                   # defines Foo action types and all Foo related action classes
+|   |   |   |       ...
+|   |   |   |
+|   |   |   +---effects                              # NGRX effects: classes that handle specific actions and react to those
+|   |   |   |       index.ts                         # barrel: re-export all effects
+|   |   |   |       foo.effects.ts                   # effects class that in reaction to actions it can interact with other services, dispatch other actions, ...
```
Comment (Mallikki) [inline showcase/README.md]: maybe "effects class that can interact with other in reaction to actions"?

File: showcase/README.md [hunk 2]
```diff
@@ -2,96 +2,149 @@
...
+|   |   |   |
+|   |   |   +---components                           # dumb components for this module
+|   |   |   |   |   index.ts                         # barrel: re-export all dumb components
+|   |   |   |   |
+|   |   |   |   +---baz                              # example dumb component
+|   |   |   |   |       baz.component.css            # styles for this dumb component
+|   |   |   |   |       baz.component.html           # template for this dumb component
+|   |   |   |   |       baz.component.spec.ts        # unit tests for this dumb component
+|   |   |   |   |       baz.component.ts             # component class for this dumb component
+|   |   |   |   |       baz.e2e.ts                   # end-to-end test for this dumb component
```
Comment (Mallikki) [inline showcase/README.md]: tests* ;)

File: showcase/README.md [hunk 3]
```diff
@@ -2,96 +2,149 @@
...
-|   |   \
+|   |       ...
 |   |
 |   |   custom-typings.d.ts             # add your own type definitions that can't be found in the registry in this file
-|   |   hmr.ts                          # Hot Module Replacement logic
 |   |   index.html                      # the main HTML page that is served when someone visits this site
-|   |   main.browser.ts                 # Angular bootstrapping
-|   |   polyfills.browser.ts            # this file includes polyfills needed by Angular and should be loaded before the app
+|   |   main.browser.ts                 # main bundle entry point, Angular bootstrapping
+|   |   polyfills.browser.ts            # polyfills needed by old browsers if needed. Should be loaded before the app
```
Comment (Mallikki) [inline showcase/README.md]: maybe "polyfills needed by old browsers, if required"? To avoid repetition?

File: showcase/README.md (hunk 1 above)
Comment (christophercr) [reply showcase/README.md]: Indeed, much better. I'll fix it ;)

File: showcase/README.md (hunk 3 above)
Comment (christophercr) [reply showcase/README.md]: True, i'll fix it

**Pull request merged**